carInsuranceBackEnd/src/main/java/com/yakvel/carInsuranceBackEnd/jwt/JwtService.java

```diff
@@ -0,0 +1,75 @@
+package com.yakvel.carInsuranceBackEnd.jwt;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.Keys;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Service;
+
+import java.security.Key;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+@Service
+public class JwtService {
+
+    @Value("${secret.key}")
+    private String SECRET_KEY;
+    public String extractUsername(String token) {
+        return extractClaim(token, Claims::getSubject);
```

Comment:
why do we need lamda here?
the code bellow will have the same impact but will use less obscure style.
```suggestion
        return extractClaim(token).getSubject();
```